dotnet/src/SemanticKernel/Memory/Collections/TopNCollection.cs

```diff
@@ -62,7 +62,7 @@ public void Add(ScoredValue<T> value)
         this._heap.Add(value);
```

Comment:
General comment: I know we looked at this once before, but this might be a good opportunity to bring in the PriorityQueue to replace TopNCollection. It's only available on .net6+ so we'd have to bring it in as a code copy for builds that are below .net6, with a compiler directive to ignore the code if net6+ (we have other examples of this).

Then we could entirely remove the whole "Collections" folder.

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

Reply:
I tried to copy `PriorityQueue` implementation, but it appeared as non-trivial task, because this data structure depends on .NET internal helpers (e.g. `EnumerableHelpers`), which should be copied as well. Also, there is a big list of warnings generated by our convention rules, and it will take some time to resolve them all (`PriorityQueue` has near 1000 lines of code).
I would probably leave `MinHeap` as it is right now.

Reply:
> e.g. EnumerableHelpers

FWIW, that's just the implementation of Enumerable.ToArray, which PQ can't rely on in the actual implementation because of layering (System.Linq.dll depends on System.Collections.dll, so the latter can't reference the former).